src/webui/www/private/scripts/mocha-init.js

```diff
@@ -384,22 +384,34 @@ const initializeWindows = function() {
         }
     };
 
-    deleteFN = function(deleteFiles = false) {
+    deleteFN = function(forceDeleteFiles = false) {
         const hashes = torrentsTable.selectedRowsIds();
-        if (hashes.length) {
+        if (hashes.length > 0) {
+            window.qBittorrent.Client.closeWindow("confirmDeletionPage");
```

Comment:
This is necessary because we no longer use iframes to load this window. Ideally, if windows reuse the same file, they should be updated and not recreated from scratch but I couldn't get it to work so for now I'm leaving it like this.

Reply:
Is it necessary to close them? Does it means that only one window of the same type is allowed at a time?
I don't think this limitation is necessary. If possible, we can allow multiple of them at the same time.

Reply:
XHR window content gets inserted directly into current document. In this particular case, we must get rid of it completely before opening another window reusing 'confirmdeletion.html' file, because otherwise we would get duplicate IDs in the DOM. There is MUI.updateContent available to replace the content of a window but either I'm using it incorrectly or it just doesn't fully work. 

Right now, you can't open the exact same window more than once no matter the method (eg. you can't edit two categories or set location of two torrents at the same time). Could this be solved? Most likely yes, but to be frank, I didn't investigate it (and nobody else wanted to do either so far). GUI only allows one dialog at a time (in most cases) and while we don't need to follow it to the letter, I'm not sure in general why more than one window "of the same type" at a time would be really needed too.

I thought using XHR is better because you need slightly less code/markup, you don't need to load styles / scripts every time on load and passing data is "easier".

Reply:
XHR window content gets inserted directly into current document. There can be only one window using 'confirmdeletion.html' at a time, because otherwise we would get duplicate IDs in the DOM.

It would be ideal to just update confirm deletion dialog when needed - MUI.updateContent is available to replace the content of a window but either I'm using it incorrectly or it just doesn't fully work. Thus, we need to reopen it so changes are reflected (eg. user selects torrent, presses Del but then selects another and presses Shift+Del). This is actually additional improvement, because previously once you opened delete window you were stuck with it (it wasn't updated).

Right now, you can't open the same window more than once no matter the method (Mocha only allows one window per ID, eg. you can't edit two categories or set location of two torrents at the same time). Could this be solved? Most likely yes, but to be frank, I didn't investigate it (and nobody else wanted to do either so far). GUI only allows one dialog at a time (in most cases) and while we don't need to follow it to the letter, I'm not sure in general why more than one window "of the same type" at a time would be really needed too.

PS. I did my best trying to explain it but I probably made it worse... :P

Reply:
>PS. I did my best trying to explain it but I probably made it worse... :P

It does provide some helpful insights. 

>Right now, you can't open the same window more than once no matter the method (Mocha only allows one window per ID, eg. you can't edit two categories or set location of two torrents at the same time). Could this be solved? Most likely yes, but to be frank, I didn't investigate it (and nobody else wanted to do either so far). GUI only allows one dialog at a time (in most cases) and while we don't need to follow it to the letter, I'm not sure in general why more than one window "of the same type" at a time would be really needed too.

I was talking generally. For this PR, yes, the confirm deletion dialog should only be one on screen at a time, i.e. it is a modal dialog.
IMO this is related to the modal/non-modal property of a dialog used in GUI. I believe the qbt GUI already has correct modal property for all dialogs, webui can just follow/emulate the behavior.

>It would be ideal to just update confirm deletion dialog when needed - MUI.updateContent is available to replace the content of a window but either I'm using it incorrectly or it just doesn't fully work. Thus, we need to reopen it so changes are reflected (eg. user selects torrent, presses Del but then selects another and presses Shift+Del). This is actually additional improvement, because previously once you opened delete window you were stuck with it (it wasn't updated).

In GUI, a modal dialog would require user attention. Unless the user accept/dismiss the dialog, the user can't do anything else, not even accessing the parent window. 
I'm not actually sure that updating the existing dialog is better but I don't strongly oppose it. Ideally I would hope the webUI would closely follow GUI behavior.

Reply:
I looked around and there is MUI.Modal actually but it looks like it has some minor overlay bug. I'll experiment with it but I would like to keep current PR like it is now if it's OK (I have another PR coming once this is merged).

Reply:
>I would like to keep current PR like it is now if it's OK 

Sure, it is already approved.

>I have another PR with further improvements coming once this is merged).

No problem. FYI, capacity of review is not unlimited. Too many PR at a time would just overload/distract the reviewer. Having 2~3 PR at most at the same time would be the optimal size from my experience.

Reply:
> IMO this is related to the modal/non-modal property of a dialog used in GUI. I believe the qbt GUI already has correct modal property for all dialogs, webui can just follow/emulate the behavior.

MUI.Modal actually works pretty decently - I'm going to update windows to use it, but probably piece by piece, here and there. It would be too much for just one PR probably.

> FYI, capacity of review is not unlimited. Too many PR at a time would just overload/distract the reviewer.

I'm sorry, it wasn't my intention to rush / overwhelm you - I just never contributed to anything before and still learning the etiquette :] WebUI is super behind and there are many small things that are fixable quickly but, I'll try not to do too many things at once.

Reply:
>MUI.Modal actually works pretty decently - I'm going to update windows to use it, but probably piece by piece, here and there. It would be too much for just one PR probably.

That would be nice.

>I'm sorry, it wasn't my intention to rush / overwhelm you - I've just never contributed to any repo before this and still learning the etiquette :] WebUI is super behind and there are many small things that are fixable quickly but, I'll try not to do too many things at once.

No need to. It seems that you will be staying here for a while so I thought it would be best to be clear about the review process. Your work is greatly appreciated!